### ISSUE TITLE
Able to read light status but unable to control light

### ISSUE THREAD

**Issue by Leytronix** [andrei-tatar/node-red-contrib-smartnora#152]
Hi,

With Nora Smart I am unable to control my device (light). If I manually turn the light On/Off the status correctly changes on the site but if I want to switch the light on/off via the site nothing happens!

Thanks,

Tjerk

My user id is: gL6Ut9EVIaYUeUfqGDVWeRA9Uaz1

**Comment by andrei-tatar** [comment]
@Leytronix Hi,
Can you share your node-red flow? What device are you using? How are you trying to control it?

**Comment by Leytronix** [comment]
Please find my flow below. The lamp I want to control is a Shelly Dimmer 2.

Thanks,

Tjerk

[
    {
        "id": "a223d0581cd334c7",
        "type": "tab",
        "label": "Flow 1",
        "disabled": false,
        "info": "",
        "env": []
    },
    {
        "id": "bed9e5c9ad9f7d39",
        "type": "inject",
        "z": "a223d0581cd334c7",
        "name": "Off",
        "props": [
            {
                "p": "payload"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "",
        "payload": "0",
        "payloadType": "str",
        "x": 230,
        "y": 2700,
        "wires": [
            [
                "151dd1e37c9fcbc0"
            ]
        ]
    },
    {
        "id": "9c68dc902a1b8323",
        "type": "debug",
        "z": "a223d0581cd334c7",
        "name": "",
        "active": true,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "payload",
        "targetType": "msg",
        "statusVal": "",
        "statusType": "auto",
        "x": 610,
        "y": 2660,
        "wires": []
    },
    {
        "id": "6faa39a2e4ea88b8",
        "type": "inject",
        "z": "a223d0581cd334c7",
        "name": "On",
        "props": [
            {
                "p": "payload"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "",
        "payload": "1",
        "payloadType": "str",
        "x": 230,
        "y": 2660,
        "wires": [
            [
                "151dd1e37c9fcbc0"
            ]
        ]
    },
    {
        "id": "151dd1e37c9fcbc0",
        "type": "noraf-light",
        "z": "a223d0581cd334c7",
        "devicename": "Kantoor lamp",
        "lightcolor": false,
        "brightnesscontrol": false,
        "commandonlycolor": false,
        "turnonwhenbrightnesschanges": false,
        "passthru": false,
        "errorifstateunchaged": false,
        "statepayload": true,
        "brightnessoverride": "",
        "roomhint": "Kantoor",
        "name": "Kantoor lamp",
        "colortype": "hsv",
        "nora": "44603155b3c4e596",
        "topic": "",
        "onvalue": "1",
        "onvalueType": "str",
        "offvalue": "0",
        "offvalueType": "str",
        "temperaturemin": "2700",
        "temperaturemax": "5500",
        "twofactor": "off",
        "twofactorpin": "",
        "filter": false,
        "x": 420,
        "y": 2660,
        "wires": [
            [
                "9c68dc902a1b8323"
            ]
        ]
    },
    {
        "id": "44603155b3c4e596",
        "type": "noraf-config",
        "name": "nora config",
        "group": "",
        "twofactor": "off",
        "twofactorpin": "",
        "localexecution": true,
        "structure": "Tjerks home",
        "storeStateInContext": true
    }
]

**Comment by andrei-tatar** [comment]
So what doesn't work for you? You should be getting 1/0 in the debug panel when you turn on/off the light. 
![image](https://user-images.githubusercontent.com/6473183/149732990-14be5fa6-7322-4044-91ed-4f4adc93b8c6.png)

The rest of the work (connecting the output of the node to your shelly lamp) is outside NORA's responsibility and is dependent on each device. I personally don't use shelly, but I can share with you how I integrate with zigbee2mqtt. 

In fact I plan to make a bunch of videos to show how to use NORA in various scenarios.

**Comment by Leytronix** [comment]
Yes, in Node-red turning on/off is working fine but the status of the light does not change both on the NORA site and also not the actual status of the lamp. However, if I turn on/off the light from Google Home the status changes correctly in both Node-red and on the NORA site! So, it looks there is only one way communication at the moment.... 

![Lamp](https://user-images.githubusercontent.com/35609073/149734658-9d94e981-61b3-4c3f-8e1f-39153c890667.png)

**Comment by andrei-tatar** [comment]
@Leytronix do you have any issues with the internet connectivity to node-red. Do you have any error messages in the node-red logs?

You can see the logs with: ```journalctl -u nodered.service``` (if running on linux/rpi as a service)

**Comment by Leytronix** [comment]
Hi Andre,

As far as I can see there is nothing wrong with the internet connection to Node-red. See output below.

Kind regards,

Tjerk

-- Logs begin at Thu 2019-02-14 11:11:59 CET, end at Mon 2022-01-17 09:38:22 CET
jan 15 09:34:45 raspberrypi systemd[1]: Started Node-RED graphical event wiring 
jan 15 09:34:48 raspberrypi Node-RED[2582]: 15 Jan 09:34:48 - [info]
jan 15 09:34:48 raspberrypi Node-RED[2582]: Welcome to Node-RED
jan 15 09:34:48 raspberrypi Node-RED[2582]: ===================
jan 15 09:34:48 raspberrypi Node-RED[2582]: 15 Jan 09:34:48 - [info] Node-RED ve
jan 15 09:34:48 raspberrypi Node-RED[2582]: 15 Jan 09:34:48 - [info] Node.js  ve
jan 15 09:34:48 raspberrypi Node-RED[2582]: 15 Jan 09:34:48 - [info] Linux 5.10.
jan 15 09:34:49 raspberrypi Node-RED[2582]: 15 Jan 09:34:49 - [info] Loading pal
jan 15 09:34:51 raspberrypi Node-RED[2582]: 15 Jan 09:34:51 - [info] Settings fi
jan 15 09:34:51 raspberrypi Node-RED[2582]: 15 Jan 09:34:51 - [info] Context sto
jan 15 09:34:51 raspberrypi Node-RED[2582]: 15 Jan 09:34:51 - [info] User direct
jan 15 09:34:51 raspberrypi Node-RED[2582]: 15 Jan 09:34:51 - [warn] Projects di
jan 15 09:34:51 raspberrypi Node-RED[2582]: 15 Jan 09:34:51 - [info] Flows file 
jan 15 09:34:51 raspberrypi Node-RED[2582]: 15 Jan 09:34:51 - [info] Creating ne
jan 15 09:34:51 raspberrypi Node-RED[2582]: 15 Jan 09:34:51 - [warn]
jan 15 09:34:51 raspberrypi Node-RED[2582]: ------------------------------------
jan 15 09:34:51 raspberrypi Node-RED[2582]: Your flow credentials file is encryp
jan 15 09:34:51 raspberrypi Node-RED[2582]: If the system-generated key is lost 
jan 15 09:34:51 raspberrypi Node-RED[2582]: file will not be recoverable, you wi
jan 15 09:34:51 raspberrypi Node-RED[2582]: your credentials.
jan 15 09:34:51 raspberrypi Node-RED[2582]: You should set your own key using th

**Comment by andrei-tatar** [comment]
@Leytronix that is only the beginning of the log. Can you scroll down (page-down/arrow keys) and check if there's anything relevant?

**Comment by Leytronix** [comment]
Ok, is there a way to save the complete log to a file?

**Comment by andrei-tatar** [comment]
Something like this should work:
```journalctl -u nodered.service --no-tail > my-log-file.log```

**Comment by drbeat** [comment]
Looks like the same problem I had on my Pi.
Unfortunately I could not make it work. But I am still curious if someone can make it work.

**Comment by drbeat** [comment]
As far as I understood from https://github.com/andrei-tatar/node-red-contrib-smartnora/issues/152#issue-1105440462 and also how It works for me:
* Light can be set On/Off via Node-Red Dashboard and status is correctly shown on smart-nora.eu
* status cannot be changed via smart-nora.eu
* Google Assistant can Trigger status but the app show light is offline

**Comment by Leytronix** [comment]
- Status of the light is shown correctly both on Google and smart-nora.eu. 

- Trigger via Google Home works but trigger via smart-nora.eu not ;-(

**Comment by andrei-tatar** [comment]
> Trigger via Google Home works but trigger via smart-nora.eu not ;-(

Interesting, I don't see any errors in the logs. Are you sure you are logged in with the same account you use in node-red?

**Comment by andrei-tatar** [comment]
> As far as I understood from [#152 (comment)](https://github.com/andrei-tatar/node-red-contrib-smartnora/issues/152#issue-1105440462) and also how It works for me:
> 
> * Light can be set On/Off via Node-Red Dashboard and status is correctly shown on smart-nora.eu
> * status cannot be changed via smart-nora.eu
> * Google Assistant can Trigger status but the app show light is offline

@drbeat what is your user id?

**Comment by Leytronix** [comment]
No, current status of the light is shown correctly both on Google and smart-nora.eu but the status can only be changed from Google Home and NOT from either Node-red or smart-nora.eu

**Comment by andrei-tatar** [comment]
@Leytronix can you try running node-red on a different machine to check if you still have this issue?
Can you give more details about how you're running node-red right now (machine/OS/node.js/npm versions)?

**Comment by Leytronix** [comment]
Now have tested it on a PC with the same result as on the pi. On smart-nora.eu I now am seeing two lights with the same name.

![afbeelding](https://user-images.githubusercontent.com/35609073/149880207-f4688143-6044-4129-922e-e656d8624756.png)

**Comment by andrei-tatar** [comment]
@Leytronix when using different instances of node-red, with the same NORA account, you should use different group names in the config node: 

![image](https://user-images.githubusercontent.com/6473183/149916676-80a9fd0c-c5b9-4dc3-8559-ddfb452f2a99.png)

**Comment by Leytronix** [comment]
Yes I know...

**Comment by Leytronix** [comment]
But in that case it is not linked to the same Google Home name anymore.

**Comment by andrei-tatar** [comment]
@Leytronix I can't think of any reason why node-red and dashboard would not be able to change the state, especially as the dashboard uses the same execution path as Google cloud.

Are you behind some kind of proxy/firewall/VPN?
Can you join slack? (https://join.slack.com/t/smart-nora/shared_invite/zt-11s3quyd2-K9~ZdvcmSK0j4wyoS1OeIQ) maybe we can schedule a call/screen sharing session to try and figure it out.

**Comment by Leytronix** [comment]
I am using NordVPN but with the VPN turned off I see the same behaviour. We can do the meeting tonight (20.00 GMT)

**Comment by Leytronix** [comment]
I still had some other flows in my workspace and have removed them. Now nora-smart.eu responds correctly to node-red! However, the light still does not respond ;-(

**Comment by andrei-tatar** [comment]
> We can do the meeting tonight (20.00 GMT)

If you can join the Slack group, we can coordinate there. 20:00 GMT is pretty late for me (22:00 EET).

**Comment by Leytronix** [comment]
Ok, I now have joined slack ;-)

**Comment by drbeat** [comment]
So I installed Node-Red on my Windows Machine.
But the behaviour is the same. I am not using a VPN oder anything inbetween. But my ISP blocks connection from outside of my network.
Could that be related?

My user id is: Hn3i6PEY2tdmjbx7JB6bVWyjPFy1

**Comment by drbeat** [comment]
Ok I had a suspect that firebase <-> node-red sync is failing due to different timezone configurations.

So on the Pi I had to make sure that the timezone matches my local timezone.
"Europe/Berlin" and you can check via "date" command if the Pi matches your PC time.
If it doesn't and the timezone is set correctly you need to change the NTP Server as it reports the wrong time to your Pi.
I had to set it to "europe.pool.ntp.org" to get it to report the correct time.

Bonus:
If you are running node-red in a docker  like me you need to make sure docker gets the correct timezone via an env variable
TZ=Europe/Berline

Now it is working for me

**Comment by Leytronix** [comment]
I now am running node-red directly on a PC with the correct time but still have one way communication via NORA ;-(
